### PR TITLE
Task/clean code vm text

### PR DESCRIPTION
**Pull request by gRaytan**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/104" target="_blank">ReviewNinja

File: app/build.gradle [hunk 1]
```diff
@@ -14,6 +14,10 @@ android {
             storeFile file('../keystores/gimelgimel.jks')
         }
     }
+    dexOptions {
```
Comment (teamagam3) [inline app/build.gradle]: Unnecessary for all of us, should be used locally

File: app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Inject
+    SendMessagePresenter sendMessagePresenter;
+
+    private Subscription subscription;
+
+    private boolean isInputValid(String userMessage) {
+        return !userMessage.isEmpty();
+    }
+    private class handlePresenterView implements SendMessagePresenter.View {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java]: The view should implements these methods.
The view model shouldn't do UI tasks like toast.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java [hunk 3]
```diff
@@ -105,41 +81,19 @@ protected Object castInterface(Fragment fragment) {
...
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        DaggerMessagesComponent.builder()
+                .applicationComponent(((GGApplication) getActivity().getApplication()).getApplicationComponent())
+                .messageModule(new MessageModule())
+                .build()
+                .inject(this);
     }
 
-    private class handlePresenterView implements SendMessagePresenter.View{
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java]: Restore this class (Already explained why).

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java [hunk 4]
```diff
@@ -2,67 +2,43 @@
...
-                    .getApplicationContext()));
-            sendMessagePresenter.sendMessage(userMessage);
-
-//            mMessageSender.sendTextMessageAsync(userMessage);
-            dismiss();
-        }
-
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mBinding = DataBindingUtil.inflate(LayoutInflater.from(getActivity()), R.layout.dialog_send_message, null, false);
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/dialogs/SendMessageDialogFragment.java]: What's the difference between DataBindingUtil.inflate() and DataBindingUtils.setContentView()?

there is no use of setContentView

File: app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+        final Context mContext;
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java]: Context is part of the view, this is another reason why this class has to be moved to the fragment.

right. no need

File: app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Override
+    public void destroy() {
+        if (subscription != null && !subscription.isUnsubscribed()) subscription.unsubscribe();
+        subscription = null;
+        this.activity = null;
+    }
+
+
+    public boolean onPositiveClick() {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/viewModels/SendMessageDialogFragmentViewModel.java]: Rename the method to better name that describes its goal and move it to the ViewModel (The dialog has this method too).

File: app/src/main/java/com/teamagam/gimelgimel/app/viewModels/ViewModel.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.teamagam.gimelgimel.app.viewModels;
+
+
+/**
+ * Created by Gil on 18/08/2016.
+ */
+public interface ViewModel {
+
+    void destroy();
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/viewModels/ViewModel.java]: Why only destroy()?

There is no common functionality yet. we will edit after merge and other UC

File: app/src/main/res/layout/dialog_send_message.xml [hunk 8]
```diff
@@ -1,15 +1,26 @@
...
-        android:id="@+id/dialog_send_geo_message_edit_text"
-        android:layout_width="fill_parent"
-        android:layout_height="wrap_content"
-        android:paddingTop="20dp"
-        android:paddingLeft="20dp"
-        android:hint="Enter your message here..."
-        android:inputType="text" />
-</LinearLayout>
\ No newline at end of file
+    <data>
```
Comment (teamagam3) [inline app/src/main/res/layout/dialog_send_message.xml]: The data on binding is the model the view works with.
The view model is just another bridge between the model and the view that updated the model by the view actions.

Didnt understand your comment

File: xwalk_core_library/build.gradle [hunk 9]
```diff
@@ -8,7 +8,10 @@ android {
         minSdkVersion 14
         targetSdkVersion 21
     }
-
+    dexOptions {
+        // Prevent OutOfMemory with MultiDex during the build phase
+        javaMaxHeapSize "4g"
```
Comment (teamagam3) [inline xwalk_core_library/build.gradle]: Same same.

File: data/build.gradle [hunk 10]
```diff
@@ -12,7 +12,10 @@ targetCompatibility = 1.8
 android {
     compileSdkVersion 23
     buildToolsVersion "23.0.2"
-
+    dexOptions {
+        // Prevent OutOfMemory with MultiDex during the build phase
+        javaMaxHeapSize "4g"
```
Comment (teamagam3) [inline data/build.gradle]: Same same.

File: app/src/main/res/layout/dialog_send_message.xml (hunk 8 above)
Comment (teamagam3) [reply app/src/main/res/layout/dialog_send_message.xml]: Nevermind, we talked about our architecture, and the binding data will be the view model.

**Pull request closed without merging**